### PR TITLE
Better deletions on frontend

### LINKED ISSUE #723: Deleting organizations and projects at frontend
For now we cannot delete projects and organizations. We need to 
- [x] Improve `updateProject` and `updateOrganization` methods in order to be able to mark project or organization as `DELETED`.
- [x] Implement deleting at frontend

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR closes #723 

### What's done:
- [x] Implemented functions to delete organizations and projects: now buttons work fine
- [ ] Add tests on ProjectView deletion
- [ ] Add tests on OrganizationView deletion

### CODE REVIEW

**Comment by akuleshov7** [comment]
Pls write some test?

**Comment by sanyavertolet** [comment]
> Pls write some test?

Now having this type of error with frontend tests for `ProjectView`:
```
debug.js:21 Error: Uncaught TypeError: jquery(...).tooltip is not a function (http://localhost:9876/absolute/var/folders/wg/stk4v7wx0h92r3p2738g6_9w0000gn/T/_karma_webpack_558967/commons.js:113564)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4210:1)
    at invokeGuardedCallback (react-dom.development.js:4274:1)
    at reportUncaughtErrorInDEV (react-dom.development.js:22737:1)
    at captureCommitPhaseError (react-dom.development.js:27078:1)
    at commitPassiveMountEffects_complete (react-dom.development.js:24783:1)
    at commitPassiveMountEffects_begin (react-dom.development.js:24768:1)
    at commitPassiveMountEffects (react-dom.development.js:24756:1)
    at flushPassiveEffectsImpl (react-dom.development.js:26990:1)
    at flushPassiveEffects (react-dom.development.js:26935:1)
```
Trying to fix it.

**Review by petertrr — APPROVED** [review]

File: save-frontend/src/test/kotlin/com/saveourtool/save/frontend/components/views/OrganizationViewTest.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        worker.resetHandlers()
+        // todo: stop worker in `afterall`. There doesn't seem to be immediate support for this in kotlin.test for JS.
+        // Possible solution is to use https://kotest.io/docs/framework/lifecycle-hooks.html
+        // worker.stop()
+    }
+
+    @Test
+    fun shouldShowConfirmationWindowWhenDeletingOrganization(): Promise<Unit> {
+        renderOrganizationView()
+        return screen.findByText("SETTINGS").then {
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/test/kotlin/com/saveourtool/save/frontend/components/views/OrganizationViewTest.kt:55]: This piece of code doesn't work fine. Seems like there is some time needed to render `organizationSettingsMenu`.
```
screen.getByText("SETTINGS").let {
    userEvent.click(it)
}
screen.getByText("Delete organization").let {
    userEvent.click(it)
}
return screen.findByText("Ok").then {
    assertNotNull(it, "Should show confirmation window")
}
```
So I will use `getBy...` only to find `SETTINGS` button.

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RequestUtils.kt [hunk 2]
```diff
@@ -85,6 +85,19 @@ suspend fun Component<*, *>.post(url: String,
                                  responseHandler: (Response) -> Unit = this::classComponentResponseHandler,
 ) = request(url, "POST", headers, body, responseHandler = responseHandler)
 
+/**
+ * Perform DELETE request from a class component. See [request] for parameter description.
+ *
+ * @return [Response]
+ */
+@Suppress("KDOC_WITHOUT_PARAM_TAG")
+suspend fun Component<*, *>.delete(
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RequestUtils.kt:94]: There already is one:
```
/**
 * Perform DELETE request from a functional component
 *
 * @return [Response]
 */
@Suppress("EXTENSION_FUNCTION_WITH_CLASS", "KDOC_WITHOUT_PARAM_TAG")
suspend fun WithRequestStatusContext.delete(
    url: String,
    headers: Headers,
    body: dynamic,
    responseHandler: (Response) -> Unit = this::withModalResponseHandler,
) = request(url, "DELETE", headers, body, responseHandler = responseHandler)
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt [hunk 3]
```diff
@@ -676,7 +676,7 @@ class OrganizationView : AbstractView<OrganizationProps, OrganizationViewState>(
         }
         scope.launch {
             responseFromDeleteOrganization =
-                    post("$apiUrl/organization/${props.organizationName}/update", headers, Json.encodeToString(state.organization))
+                    delete("$apiUrl/organization/${props.organizationName}/delete", headers, "")
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt]: Done

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RequestUtils.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (petertrr) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RequestUtils.kt:94]: Oh, ok

**Pull request merged**